### PR TITLE
Update wavefront_obj crate for faster obj asset loading

### PR DESCRIPTION
**Pull request by Frizi**
## Description

This updates the `wavefront_obj` crate in order to include latest optimization changes from https://github.com/PistonDevelopers/wavefront_obj/pull/57

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [ ] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [ ] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ The format is based on [Keep a Changelog][kc], and this project adheres to
 
 ### Added
 
+* Optimize loading of wavefront obj mesh assets by getting rid of unnecessary allocations. ([#1454])
```
Comment (jojolepro) [inline docs/CHANGELOG.md]: Shouldn't that be in ### Fixed ?

Comment (Frizi) [reply docs/CHANGELOG.md]: I don't know, it wasn't a bug really.

Comment (Frizi) [reply docs/CHANGELOG.md]: Anyway, i'll move it

**Comment by Moxinilian** [comment]
Shall we get this merge right now?

**Comment by jojolepro** [comment]
bors r=Jojolepro,Moxinilian

**Comment by Moxinilian** [comment]
bors retry

**Pull request merged**